### PR TITLE
Kafka source: Use Segment's kafka-go client

### PR DESCRIPTION
**Pull request by hariso**
### Description

List of major changes:
1. The Kafka source connector now uses Segment's Kafka client.
2. Test packages are now `kafka_test`. This is because the tests (which were in the `kafka` package), used the `mock` package, which now imports something from `kafka` package, which created an import cycle Go complained about.
3. Offsets are now managed by the Kafka brokers. In the current plugin interfaces, we have the `Read()` method, which should make it possible to read a record from an arbitrary position. To achieve that with Confluent's Kafka client, we were manually managing offsets. Practically, though, never needed to read records from arbitrary positions. Furthermore, in the new plugin interfaces, we won't have to do that at all: the plugin will stream messages to Conduit.

I'm about to update the plugin docs in a bit, and also do a bit more of cleanup.

Fixes part of #52 

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [x] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed. -- Well, kinda. I'm not sure if the PR can be made smaller.

### CODE REVIEW

**Review by dylanlott — APPROVED** [review]
I was able to build and run the docker mage locally and received messages into my local Kafka from a file source.

When I was building the docker image, I compared the current image size of `main` to this version and this PR reduces the final build image size by ~79 MB. 👍 🍾

File: pkg/plugins/kafka/destination_integration_test.go [hunk 1]
```diff
@@ -24,24 +24,21 @@ import (
...
 	"github.com/conduitio/conduit/pkg/foundation/assert"
 	"github.com/conduitio/conduit/pkg/plugins"
+	"github.com/conduitio/conduit/pkg/plugins/kafka"
 	"github.com/conduitio/conduit/pkg/record"
-	"github.com/confluentinc/confluent-kafka-go/kafka"
 	"github.com/google/uuid"
+	skafka "github.com/segmentio/kafka-go"
 )
 
 // todo try optimizing, the test takes 15 seconds to run!
```
[the remark this answers is not in the document]
Comment (hariso) [reply pkg/plugins/kafka/destination_integration_test.go:33]: That appears to be the case. From what I could understand from logs, tearing down (i.e. simply closing the client), takes cca. 5 seconds.:S

File: pkg/plugins/kafka/source_test.go [hunk 2]
```diff
@@ -52,163 +52,90 @@ func TestTeardownSource_ClosesClient(t *testing.T) {
...
 		EXPECT().
-		Get(msgTimeout).
-		Return(kafkaMsg, expPos, nil)
+		Get(gomock.Any()).
+		Return(kafkaMsg, groupID, nil)
 
-	underTest := Source{Consumer: consumerMock, Config: cfg}
-	rec, err := underTest.Read(context.TODO(), pos)
+	underTest := kafka.Source{Consumer: consumerMock, Config: cfg}
+	rec, err := underTest.Read(context.TODO(), []byte(groupID))
```
[the remark this answers is not in the document]
Comment (hariso) [reply pkg/plugins/kafka/source_test.go:76]: Good point, I'll change this.

File: pkg/plugins/kafka/config.go [hunk 3]
```diff
@@ -39,46 +38,31 @@ var Required = []string{Servers, Topic}
...
-	Servers string
+	Servers []string
 	Topic   string
-	// Maps to "security.protocol" in a Kafka consumer's configuration
-	SecurityProtocol string
-	// Maps to "acks" in a Kafka consumer's configuration
-	Acks            skafka.RequiredAcks
+	// Required acknowledgments when writing messages to a topic:
+	// Can be: 0, 1, -1 (all)
+	Acks            kafka.RequiredAcks
```
Comment (ahmeroxa) [inline pkg/plugins/kafka/config.go:45]: Are we going to make this configurable?

Comment (ahmeroxa) [reply pkg/plugins/kafka/config.go:45]: Oh wait, it is already nm

**Comment by ahmeroxa** [comment]
I'm confused by how the GroupID is being generated/handled here. It looks like we're using the `position` as the GroupID.

GroupID should be configurable (since a customer might want to use a specific consumer group ID). If it's not set, then we can generate one internally but it needs to be remain consistent throughout the entire lifespan of the connector.

**Comment by hariso** [comment]
> I'm confused by how the GroupID is being generated/handled here. It looks like we're using the `position` as the GroupID.
> 
> GroupID should be configurable (since a customer might want to use a specific consumer group ID). If it's not set, then we can generate one internally but it needs to be remain consistent throughout the entire lifespan of the connector.

We talked about this offline, but for the rest: Group ID currently is generated the first time a source is read, and then re-used throughout a connector's lifespan. Ali mentioned that some Kafka deployments will use fixed consumer groups (e.g. DevOps will explicitly create Groups with certain ACLs etc.), so I'll make the group ID configurable.

**Review by ahmeroxa — APPROVED** [review]
LGTM 👍

**Pull request merged**